main.go

```diff
@@ -72,9 +72,10 @@ func Stub(w http.ResponseWriter, r *http.Request) {
 
 func Chunk(w http.ResponseWriter, r *http.Request) {
 	type ChunkRequestBody struct {
-		Chunk    string                 `json:"chunk"`
-		FilePath string                 `json:"filepath"`
-		MetaData map[string]interface{} `json:"metadata,omitempty"`
+		Chunk         string                 `json:"chunk"`
+		FilePath      string                 `json:"filepath"`
+		EmbeddingText *string                `json:"embedding_text,omitempty"`
```

Comment:
Stick to a consistent naming format in the json tag:
```suggestion
		EmbeddingText *string                `json:"embeddingtext,omitempty"`
```